sa/sa.go

```diff
@@ -1667,6 +1685,9 @@ func (ssa *SQLStorageAuthority) RevokeCertificate(ctx context.Context, req *sapb
 // the given identifier, if available. This method is intended to deprecate
 // GetPendingAuthorization. This method only supports DNS identifier types.
 func (ssa *SQLStorageAuthority) GetPendingAuthorization2(ctx context.Context, req *sapb.GetPendingAuthorizationRequest) (*corepb.Authorization, error) {
+	if req.RegistrationID == 0 || req.IdentifierValue == "" || req.ValidUntil == 0 {
```

Comment:
(Not suggesting you change this, but just noting: if `RegistrationID` or `IdentifierValue` are the empty strings here... nothing bad happens! The SELECT will come up empty, we'll return `NotFoundError`, and everything is fine. But since we _know_ that those zero values should only every crop up here as a result of an error, returning a 500 instead of a 400 is probably good.)

Reply:
I think preempting the execution of a query that we know won't succeed is probably a win. Returning a NotFound error here in place of `errIncompleteRequest` would be a bit funny given that we shouldn't even be willing to look this up unless the subscriber has provided the required fields.